#! /usr/bin/env python

PACKAGE='manzanita_mk3_driver'
import roslib; roslib.load_manifest(PACKAGE)

from device_driver_base.msg import SensorLevels
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

#       Name                   Type      Reconfiguration level             Description         Default   Min   Max
gen.add("port",                str_t,    SensorLevels.RECONFIGURE_STOPPED, "The port threw which to talk to the device", "/dev/ttyUSB1")

gen.add("update_rate",         double_t, SensorLevels.RECONFIGURE_OPEN, "The frequency at which updates are pushed", 0.25, 0.1, 2)

gen.add("num_parallel_units",  int_t,    SensorLevels.RECONFIGURE_OPEN, "The number of series sets of units in parallel", 2, 0, 10)
gen.add("num_series_units",    int_t,    SensorLevels.RECONFIGURE_OPEN, "The number of series units for each parallel set", 1, 0, 10)
gen.add("cells_per_unit",      int_t,    SensorLevels.RECONFIGURE_OPEN, "The number of cells in each unit", 6, 0, 12)

gen.add("battery_topic",       str_t,    SensorLevels.RECONFIGURE_OPEN, "The topic on which battery information is published", "battery")
gen.add("temperature_topic",   str_t,    SensorLevels.RECONFIGURE_OPEN, "The topic on which temperature information is publised", "temperature")

gen.add("frame_id",            str_t,    SensorLevels.RECONFIGURE_RUNNING, "The root frame of the battery", "/battery")

gen.add("max_cell_voltage",    double_t, SensorLevels.RECONFIGURE_RUNNING, "The maximum voltage that a cell can have", 4.5, 0.1, 10)
gen.add("max_total_voltage",   double_t, SensorLevels.RECONFIGURE_RUNNING, "The maximum voltage that the pack can have", 27, 0.1, 50)
gen.add("warn_total_voltage",  double_t, SensorLevels.RECONFIGURE_RUNNING, "The voltage when the node will start indicating a low battery", 27, 0.1, 50)
gen.add("critical_total_voltage",double_t, SensorLevels.RECONFIGURE_RUNNING, "The voltage when the node will start indicating a critical battery level", 27, 0.1, 50)


exit(gen.generate(PACKAGE, "manzanita_mk3_driver", "MK3"))
